packages/framework/src/Model/Product/Search/Export/ProductSearchExportWithFilterRepository.php

```diff
@@ -53,28 +53,62 @@ public function getProductsData(int $domainId, string $locale, int $startFrom, i
         $result = [];
         /** @var \Shopsys\FrameworkBundle\Model\Product\Product $product */
         foreach ($query->getResult() as $product) {
-            $flagIds = $this->extractFlags($product);
-            $categoryIds = $this->extractCategories($domainId, $product);
-            $parameters = $this->extractParameters($locale, $product);
-            $prices = $this->extractPrices($domainId, $product);
-
-            $result[] = [
-                'id' => $product->getId(),
-                'catnum' => $product->getCatnum(),
-                'partno' => $product->getPartno(),
-                'ean' => $product->getEan(),
-                'name' => $product->getName($locale),
-                'description' => $product->getDescription($domainId),
-                'shortDescription' => $product->getShortDescription($domainId),
-                'brand' => $product->getBrand() ? $product->getBrand()->getId() : '',
-                'flags' => $flagIds,
-                'categories' => $categoryIds,
-                'in_stock' => $product->getCalculatedAvailability()->getDispatchTime() === 0,
-                'prices' => $prices,
-                'parameters' => $parameters,
-                'ordering_priority' => $product->getOrderingPriority(),
-                'calculated_selling_denied' => $product->getCalculatedSellingDenied(),
-            ];
+            $result[] = $this->extractResult($product, $domainId, $locale);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param int $domainId
+     * @param string $locale
+     * @return array
+     */
+    protected function extractResult(Product $product, int $domainId, string $locale): array
```

Comment:
Probably the whole repository should be split into two classes, but I don't want to introduce this change in this PR as it will be probably a BC break

Reply:
We should prevent that in 8.0 I think... And now I am not sure if this should be part of `8.0 deprecations removement` or somewhere else, but I think we should do that.